graylog2-web-interface/src/components/graylog/Panel.jsx

```diff
@@ -7,59 +9,206 @@ import teinte from 'theme/teinte';
 
   .panel-footer {
     background-color: ${teinte.secondary.tre};
     border-top-color: ${teinte.secondary.due};
   }
-
   .panel-group {
     .panel-heading {
       + .panel-collapse > .panel-body,
       + .panel-collapse > .list-group {
         border-top-color: ${teinte.secondary.due};
       }
     }
-
     .panel-footer {
       + .panel-collapse .panel-body {
         border-bottom-color: ${teinte.secondary.due};
       }
     }
   }
 
-  ${bsStyleThemeVariant(panelVariantStyles)}
+  ${bsStyleThemeVariant(deprecatedVariantStyles)}
 `;
 
-export default StyledPanel;
+const CollapsibleBody = ({ children }) => {
+  return (
+    <DeprecatedStyledPanel.Collapse>
+      <DeprecatedStyledPanel.Body>
+        {children}
+      </DeprecatedStyledPanel.Body>
+    </DeprecatedStyledPanel.Collapse>
+  );
+};
+
+const Panel = ({
+  title,
+  children,
+  collapsible,
+  expanded,
+  footer,
+  header,
+  onToggle,
+  ...props
+}) => {
+  /** NOTE: Deprecated & should be removed in 4.0 */
+  if (header || footer || title || collapsible || typeof children === 'string') {
+    /* eslint-disable-next-line no-console */
+    console.warn('Graylog Panel: ', 'You have used a deprecated `Panel` prop, please check the documentation to use the latest props.');
+
+    return (
+      /* NOTE: this exists as a deprecated render for older Panel instances */
+      <DeprecatedStyledPanel {...props} expanded={expanded} onToggle={onToggle}>
+        {header && (
+          <DeprecatedStyledPanel.Heading>{header}</DeprecatedStyledPanel.Heading>
+        )}
+        {collapsible
+          ? <CollapsibleBody>{children}</CollapsibleBody>
+          : <DeprecatedStyledPanel.Body>{children}</DeprecatedStyledPanel.Body>}
+        {footer && (
+          <DeprecatedStyledPanel.Footer>{footer}</DeprecatedStyledPanel.Footer>
+        )}
+      </DeprecatedStyledPanel>
+    );
+  }
+
+  return (
+    <StyledPanel expanded={expanded}
+                 onToggle={onToggle}
+                 {...props}>
+      {children}
+    </StyledPanel>
+  );
+};
+
+CollapsibleBody.propTypes = {
+  children: PropTypes.any.isRequired,
+};
+
+Panel.propTypes = {
+  children: PropTypes.any.isRequired,
+  /** @deprecated No longer used, replace with `<Panel.Collapse />` &  `expanded`. */
+  collapsible: PropTypes.bool,
+  /**
+   * Controls the collapsed/expanded state ofthe Panel. Requires
+   * a `Panel.Collapse` or `<Panel.Body collapsible>` child component
+   * in order to actually animate out or in.
+   *
+   * @controllable onToggle
+   */
+  expanded: PropTypes.bool,
+  /**
+   * A callback fired when the collapse state changes.
+   *
+   * @controllable expanded
+   */
+  onToggle: PropTypes.func,
+  /** @deprecated No longer used, replace with `<Panel.Footer />`. */
+  footer: PropTypes.string,
+  /** @deprecated No longer used, replace with `<Panel.Heading />`. */
+  header: PropTypes.string,
+  /** @deprecated No longer used, replace with `<Panel.Title />`. */
```

Comment:
I couldn't find `Panel.Title` in the documentation, do you have any examples of that?

Reply:
Sure thing, there are a few examples here
https://react-bootstrap-v3.netlify.com/components/panel/#panels-heading
https://react-bootstrap-v3.netlify.com/components/panel/#panels-contextual
https://react-bootstrap-v3.netlify.com/components/panel/#panels-collapsible